### ISSUE TITLE
Add conversation view

### ISSUE THREAD

**Issue by cketti** [thundernest/k-9#6197]
In addition to the current threaded view, add support for a conversation view that displays all messages in a conversation. This is what most mobile clients do these days.

In the message list, we show one entry per conversation just like we do for threads. Only when conversation view is enabled, tapping this entry will not open an additional message list screen, but a message view screen that displays all messages, one below the other, starting with the oldest message. Messages that are already marked as read should be displayed in a collapsed state so only some basic information is visible (contact picture and display name of the author, date/time, message preview?). The view should scroll to the first unread message. Unread messages are displayed in an expanded state where the message body is visible (we should probably collapse quoted text, though).

This is what a conversation looks like in the Gmail app:

<img src="https://user-images.githubusercontent.com/218061/181304057-796cbcf9-2ffb-479a-a18d-53b7da4078ed.png" alt="image" width=300> <img src="https://user-images.githubusercontent.com/218061/181304524-b0761d4c-b601-4bc3-8e99-d89f04ef6909.png" alt="image" width=300>

Open questions: 
- How do we determine exactly which messages are part of a conversation? By default sent messages are saved to the Sent folder. So we definitely want to include those messages. What about archived messages? Drafts? Some messages might be on totally different folders. Do we care about those?
- How do we handle long conversations? Do we break them apart after a limit is reached (I believe Gmail is doing that). If not, we need a way to only load/display parts of a conversation. Show a placeholder like the Gmail app? Display only relevant parts of the "list" and lazy load the rest?

Note: This is very similar, but not the same as
- #735

**Comment by cketti** [comment]
A first draft.

Opening a conversation will collapse the previously read messages into a placeholder item.

![k9mail__message_view__conversation_with_placeholder](https://user-images.githubusercontent.com/218061/189201440-9a77b5e9-2f65-4356-8c63-5be630fa946b.png)

Expanding the placeholder shows those previous messages (in a collapsed state).

![k9mail__message_view__conversation_with_collapsed_messages](https://user-images.githubusercontent.com/218061/189201810-4d08edb4-8aba-4d7a-ae26-f6c57b8042ab.png)

**Comment by parisni** [comment]
Agreed @cketti . Having conversation spread on inbox AND sent message is not user friendly. Or at least, this approach is not my friend, but ennemy

**Comment by kevincox** [comment]
Personally I would like to see something in-between [Threaded view across folders · Issue #735) and a conversation view. I think my use case is closer to conversation view so I'll share my ideas here.

For me the use case is I have an email in my inbox and am reading it. 9/10 times I don't need any context so I don't want my inbox to show the full thread across folders and I don't want too much screen space wasted on conversation view inline. (For example GMail can push the message you are trying to view off screen in a long thread which is quite annoying). In all of the mockups posted above it seems that at least a quarter of the screen is spent on past messages which seems like a big waste to me as I can usually remember well enough how many messages they were and who they were from. The one exception is the "8 previous messages" one with who they are from. That isn't too bad but still seems like a lot of screen space for what I consider rarely useful information.

For me I would probably want a way to optionally expand a thread and see the branches fairly accessible when viewing a message. For example Thunderbird desktop has this fantastic "Open Message in Conversation" context menu item. I rarely use it but it is amazing for getting the context of a message when I need to. With just one click all of the branches of the thread are laid out before you.

So basically for me I think the current threaded view does a good job grouping unread messages from the same folder into one view for easy reading. I wouldn't want to have too much information about a thread's history on screen for any threaded message. **For me the key feature isn't always showing this context, but making it easily available.** Right now I need to go into my Trash and Sent folders and search for the conversation to gather up the history. This is very inconvenient. Having a "Open Message in Conversation" button which pulled up a list view (like a search results page) with a tree view of the thread would be absolutely perfect.

**Comment by oktayacikalin** [comment]
Is this actually in the works? I would love having a proper conversation view. The new Android app is shaping up nicely and this should not be neglected. On Linux I can install an extension, but honestly TB should do this natively.
